src/Node/FeatureNode.php

```diff
@@ -223,6 +223,26 @@ public function getLine()
         return $this->line;
     }
 
+    /**
+     * Returns a copy of this features but with a different set of scenarios.
+     *
+     * @param array<array-key, ScenarioInterface> $scenarios
+     */
+    public function withScenarios(array $scenarios): self
+    {
+        return new self(
+            $this->getTitle(),
+            $this->getDescription(),
+            $this->getTags(),
+            $this->getBackground(),
+            array_values($scenarios),
+            $this->getKeyword(),
+            $this->getLanguage(),
+            $this->getFile(),
+            $this->getLine()
```

Comment:
Personally for this kind of `clone with` function, I would reference the properties rather than the getters e.g.:

```
return new self(
  $this->title,
  $this->description,
  /// etc
);
```

It's more performant, clearer IMO, and also guarantees the cloned object has the exact same internal state (e.g. if in future a getter was given behaviour other than just returning the raw property value).

Reply:
I was considering that but then I worried it might not be backward compatible, e.g. if the end user somehow managed to swap the `$scenario` instance with their own subclass the behaviour would differ and possibly.

But then again those properties are private, so if the user supplied wrong parent-constructor-arguments it's already broken in principle.

I'll fix this as suggested, let me know what you think about the above in the meantime.

Reply:
Yeah, I think obviously it would be better if our classes were `final` in a future major, considering that the filtering process will always clone these back to our own implementations even if the user has subclassed, which could be confusing.

That said, I think it's unlikely anyone would have replaced the hardcoded parts of the `Parser` to instantiate custom classes.

Considering that the properties are all private, I think it's best to clone them direct.